Ensure logintoboggan_user_update receives the correct number of

anemirovsky opened the pull request:
arguments.

Fixes #2

anemirovsky committed:
Ensure logintoboggan_user_update receives the correct number of
arguments.

Fixes #2

anemirovsky committed:
Also fix hook_user_update in the main module.

argiepiano commented on logintoboggan.module:
```diff
@@ -1046,20 +1046,20 @@ function logintoboggan_user_load($users) {
     // Check to see if roles present, and the pre-auth role was present when
     // the form was initially displayed.
-    if (isset($edit['logintoboggan_pre_auth_check'])) {
+    if (isset($account->logintoboggan_pre_auth_check)) {
```
Here, we have a problem. The `$account` variable is a `User`object. This property `logintoboggan_pre_auth_check` does not exists in the user object, so this check will always return FALSE. 

In D7, this was part of the `$edit` array that this hook received. Now, since Backdrop doesn't have an `$edit` parameter, you'll need to attach this property to the user object itself during the form alter hook (`logintoboggan_form_user_profile_form_alter()`). So, in line 159, instead of 

```php
      $form['logintoboggan_pre_auth_check'] = array(
        '#type' => 'hidden',
        '#value' => '1',
      );
```

I would suggest:

```php
  $account->logintoboggan_pre_auth_check = TRUE
```

Please do check that this works.

anemirovsky replied:
I fixed this as you noted and also found some other issues that were preventing this functionality from working and have fixed those. Thank you!

anemirovsky committed:
Ensure the `logintoboggan_pre_auth_check` functionality works.

Now that the `$edit` argument is no longer passed to
`hook_user_update()` implementations we need a different way to pass
around the `logintoboggan_pre_auth_check` value to ensure that the
existing functionality where an "activated" email is sent to the user
when and admin removes the "unverified" user role is preserved.

While doing this, I found several other areas where code was attempting
to access array keys of `$account->roles` where Backdrop has now made
those keys just an incrementing key rather than the role id. There were
also some changes to the user profile form that changed the
`hook_form_alter()` implementation for how to access the user roles and
status fields to alter them.
